### PR TITLE
Add route waypoints in RoadWaypoints

### PR DESCRIPTION
**Pull request by liamchzh**

### CODE REVIEW

**Review by Gamenot — CHANGES_REQUESTED** [review]
Some commments.

**Review by Gamenot — APPROVED** [review]

**Review by aurorahcx — APPROVED** [review]

File: smarts/core/sensors.py [hunk 1]
```diff
@@ -966,21 +967,36 @@ def __init__(self, sim, vehicle, mission_planner, lookahead=50):
...
-        return self._mission_planner.waypoint_paths_at(
-            sim=self._sim,
-            pose=self._vehicle.pose,
-            lookahead=self._lookahead,
-            vehicle=self._vehicle,
-        )
+        waypoints_with_task = None
+        if self._mission_planner.mission.task is not None:
+            if isinstance(self._mission_planner.mission.task, UTurn):
+                waypoints_with_task = self._mission_planner.uturn_waypoints(
```
[the remark this answers is not in the document]
Comment (liamchzh) [reply smarts/core/sensors.py:973]: Won't address it for now.

File: smarts/core/sensors.py [hunk 2]
```diff
@@ -966,21 +967,36 @@ def __init__(self, sim, vehicle, mission_planner, lookahead=50):
...
+                    self._sim, self._vehicle.pose, self._vehicle
+                )
+            elif isinstance(self._mission_planner.mission.task, CutIn):
+                waypoints_with_task = self._mission_planner.cut_in_waypoints(
+                    self._sim, self._vehicle.pose, self._vehicle
+                )
+
+        if waypoints_with_task:
```
[the remark this answers is not in the document]
Comment (liamchzh) [reply smarts/core/sensors.py]: Actually `if waypoints_with_task` was my intention, waypoints_with_task can either be `None` or `[]`.

File: smarts/core/sensors.py [hunk 3]
```diff
@@ -82,6 +82,7 @@ class EgoVehicleObservation(NamedTuple):
 
 class RoadWaypoints(NamedTuple):
     lanes: Dict[str, List[Waypoint]]
+    route_waypoints: List[Waypoint]
```
Comment (liamchzh) [inline smarts/core/sensors.py:85]: @mg2015started

**Pull request merged**